test/e2e/framework/firewall_util.go

```diff
@@ -302,6 +302,72 @@ func PackProtocolsPortsFromFirewall(alloweds []*compute.FirewallAllowed) []strin
 	return protocolPorts
 }
 
+type portRange struct {
+	protocol  string
+	startPort int64
+	endPort   int64
+}
+
+// isPortsSubset asserts that the "requiredPorts" are covered by the "coverage" ports.
+// requiredPorts - must be single-port, examples: 'tcp/50', 'udp/80'.
+// coverage - single or port-range values, example: 'tcp/50', 'udp/80-1000'.
+// Returns true if every requiredPort exists in the list of coverage rules.
+func isPortsSubset(requiredPorts, coverage []string) error {
+	for _, reqPort := range requiredPorts {
```

Comment:
Can we split the parsing part out to make it easier to read

```
func parseFirewallPortRange(...) (*firewallPortRange, error) {
  ...
}

type firewallPortRange struct {
  protocol string
  // ports is [min, max], [x,x] represents a single port x
  min, max int
}
```

Reply:
Sure

Reply:
Done